spring-cloud-circuitbreaker-resilience4j/src/test/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JDefaultSemaphoreBulkheadIntegrationTest.java

```diff
@@ -0,0 +1,147 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.circuitbreaker.resilience4j;
+
+import java.time.Duration;
+
+import io.github.resilience4j.bulkhead.BulkheadConfig;
+import io.github.resilience4j.bulkhead.BulkheadRegistry;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkheadConfig;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkheadRegistry;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerOnErrorEvent;
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerOnSuccessEvent;
+import io.github.resilience4j.core.EventConsumer;
+import io.github.resilience4j.timelimiter.TimeLimiterConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.cloud.client.circuitbreaker.CircuitBreaker;
+import org.springframework.cloud.client.circuitbreaker.CircuitBreakerFactory;
+import org.springframework.cloud.client.circuitbreaker.Customizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Service;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+/**
+ * @author Robert McNees
+ */
+@SpringBootTest(webEnvironment = RANDOM_PORT,
+		classes = Resilience4JDefaultSemaphoreBulkheadIntegrationTest.Application.class,
+		properties = { "management.endpoints.web.exposure.include=*",
+				"spring.cloud.circuitbreaker.resilience4j.enableSemaphoreDefaultBulkhead=true" })
+@DirtiesContext
+public class Resilience4JDefaultSemaphoreBulkheadIntegrationTest {
+
+	@Mock
+	static EventConsumer<CircuitBreakerOnErrorEvent> slowErrorConsumer;
+
+	@Mock
+	static EventConsumer<CircuitBreakerOnSuccessEvent> slowSuccessConsumer;
+
+	@Autowired
+	Resilience4JDefaultSemaphoreBulkheadIntegrationTest.Application.DemoControllerService service;
+
+	@Autowired
+	Resilience4JConfigurationProperties configurationProperties;
+
+	@Autowired
+	private TestRestTemplate rest;
+
+	@SpyBean
```

Comment:
The new mockito-extensions file is used so that this SpyBean (and the 1 below it) can mock a final class